src/main.py

```diff
@@ -91,7 +92,18 @@ class EvalContext:
     permuters: List[Permuter] = field(default_factory=list)
 
 
-def write_candidate(perm: Permuter, result: CandidateResult) -> None:
+def trim_source(source, fn_name):
```

Comment:
```suggestion
def trim_source(source: str, fn_name: str) -> str:
```
to avoid mypy complaints